src/core/fonts.js

```diff
@@ -2642,9 +2649,21 @@ class Font {
     }
 
     font.pos = (font.start || 0) + tables.maxp.offset;
-    const version = font.getInt32();
+    let version = font.getInt32();
     const numGlyphs = font.getUint16();
 
+    if (version !== 0x00010000 && version !== 0x00005000) {
+      // https://learn.microsoft.com/en-us/typography/opentype/spec/maxp
+      if (tables.maxp.length === 6) {
+        version = 0x0005000;
+      } else if (tables.maxp.length >= 22) {
```

Comment:
I just took the same value we've here:
https://github.com/mozilla/pdf.js/blob/1b88aad0db542d7bc04babffa137c4662c6b971a/src/core/fonts.js#L2698
I've no idea if a font with `maxp.version == 0x10000` and an incorrect table length would be correctly rendered or not, so I thought that it'd minimize the risk of breaking something in taking 22 instead of 32.

Reply:
It looks to me like this may be a typo that originated somewhere long ago, and has just been blindly copied. Look at this code, just slightly further along at https://github.com/mozilla/pdf.js/blob/1b88aad0db542d7bc04babffa137c4662c6b971a/src/core/fonts.js#L2764-L2767

It does the same length check, but then writes to data[26] and [27], which if the length were only 22 would be out-of-bounds!

IMO that length check should always be 32, not 22.

Reply:
ah ah, yes you're right.
Let me fix it and re-run the tests.

Reply:
The 22 has been added 10 years ago:
https://github.com/mozilla/pdf.js/pull/3053
and I don't find any explanations about it so it's very likely a typo.

Reply:
Thanks for the work you started!
We also studied the project, but could not solve the problem.

I'll send you a working PDF via PM, can you please look at it too?

For our part, we’ll try to switch to your thread and give feedback whether the PDF worked or not

Reply:
@Lyumih, you can try to open the pdf with a one of the viewers here:
https://github.com/mozilla/pdf.js/pull/17253#issuecomment-1805712959
If it works it's great, if it isn't please file a new bug with a test case.